### ISSUE TITLE
Starting an app that requires HTTPS errors out

### ISSUE THREAD

**Issue by dalin-** [kalabox/kalabox#1657]
# Bug Report

**Tell us about your setup**
Repeatable in 0.13.0-rc.2, and 2.0, both on OSX

Have you previously installed Kalabox? If so, what version?
Yes, many alphas & betas, all faithfully tore down correctly

If the app requires HTTPS (i.e. redirects HTTP to HTTPS) then starting the app results in the following:

```
☼ kbox start
Starting.
Cleaning up.
Running pre start tasks.
Starting containers.
Starting episcopalassetmap_data_1
Starting episcopalassetmap_redis_1
Starting episcopalassetmap_unison_1
Starting episcopalassetmap_solr_1
Starting episcopalassetmap_web_1
Starting episcopalassetmap_db_1
Starting episcopalassetmap_cli_1
Starting episcopalassetmap_terminus_1
Starting episcopalassetmap_appserver_1
Starting episcopalassetmap_edge_1
Running post start tasks.
kalabox_proxy_1 is up-to-date
kalabox_dns_1 is up-to-date
Waiting for site to be ready.
Unhandled rejection VError: Failed after 50 retries. {"max":50,"backoff":500}: DEPTH_ZERO_SELF_SIGNED_CERT
    at /Applications/Kalabox.app/Contents/MacOS/bin/lib/promise.js.jx:70:17
    at process._tickCallback (node.js:917:13)
```

Can we either accept self-signed certs, or use something CLI to check the app (like `drush`) that won't use certs?

**Comment by pirog** [comment]
You should be able to accept the self-signed certs in the browser and your site is probably already running in the browser (despite that kalabox says it is waiting).

I think we probably just want to check sure our internal check can whitelist the self-signed certs
http://stackoverflow.com/questions/23601989/client-certificate-validation-on-server-side-depth-zero-self-signed-cert-error

**Comment by pirog** [comment]
@dalin- i just tried this and it seems to work for me. Can you provide more details on how you enforce HTTPS?

**Comment by dalin-** [comment]
@pirog 
We've got this snippet in settings.php:

```php
// Require HTTPS on dev/test/prod.
if ($conf['server_environment'] != 'local' && php_sapi_name() != "cli") {
  if (
    // If the site runs on Pantheon, or elsewhere, there may be different·
    // headers in use.  Test for both.
    isset($_SERVER['HTTP_X_SSL']) && strtoupper($_SERVER['HTTP_X_SSL']) != 'ON' ||
    isset($_SERVER['HTTPS']) && strtoupper($_SERVER['HTTPS']) != 'ON' ||
    empty($_SERVER['HTTPS']) && empty($_SERVER['HTTP_X_SSL'])
  ) {
    header('HTTP/1.0 301 Moved Permanently');
    header('Location: https://'. $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI']);
    exit();
  }
}
```

**Comment by pirog** [comment]
@dalin- hmmm weird, that was the exact snippet that i tried and it seems to work ok. Let me try again and see if i missed something.
